modules/internal/DefaultRectangular.chpl

```diff
@@ -1926,12 +1926,13 @@ module DefaultRectangular {
           yield info.theDataChunk(0)(i);
         }
       } else {
-        const stride = viewDom.ranges(1).stride: viewDom.idxType,
-              start  = viewDom.ranges(1).first,
+        const viewDomDim = viewDom.dsiDim(1),
+              stride = viewDomDim.stride: viewDom.idxType,
+              start  = viewDomDim.first,
               first  = info.getDataIndex(start),
               second = info.getDataIndex(start + stride),
               step   = (second-first):chpl__signedType(viewDom.idxType),
-              last   = first + (viewDom.ranges(1).length-1) * step:viewDom.idxType;
+              last   = first + (viewDomDim.length-1) * step:viewDom.idxType;
```

Comment:
@benharsh: Would you review that this is correct (the logical change of converting .ranges(1) calls over to .dsiDim(1) calls...  I can't figure out how this worked before, or maybe it was just never called?  My change is obfuscated somewhat by my pulling the dsiDim query into a temp, knowing that the call isn't dirt cheap in rank-change land.